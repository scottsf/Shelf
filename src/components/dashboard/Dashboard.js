import React, {Component} from 'react';
import Product from '../product/Product.js';
import './dashboard.scss';
import axios from 'axios';

class Dashboard extends Component {

  removeProduct = (id) => {
    axios.delete(`/api/product/${id}`)
      .then(res => {
        this.props.invokeGetInventory();
      })
  }

  render() {
    console.log(this.props.inventoryList)
    let inventoryList = this.props.inventory.map((item, i) => (
      <Product item={item} key={i} invokeRemoveProduct={this.removeProduct} id={item.id}/>
    ))
    return (
      <div className='dashboard'>
        {inventoryList}
      </div>
    )
  }
}

export default Dashboard;
