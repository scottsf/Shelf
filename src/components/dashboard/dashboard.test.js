import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from './Dashboard.js';
import axios from 'axios';

it('calls', () => {
  const mockProducts = [
    {
      id: 141,
      img: "http://tineye.com/images/widgets/mona.jpg",
      name: "Monnna",
      price: 9988
    }
  ]

  const wrapper = shallow(<Dashboard item={mockProducts} />);
  expect(wrapper).toMatchSnapshot();
})
// it('testest', () => {
//   const fetchData = axios.get('/api/inventory')
//     .then(resolve => {
//       console.log(resolve)
//       return resolve.data
//     });
//   console.log(dashboard)
//   expect(dashboard.getInventory(fetchData).length().toBe(2))
// })
