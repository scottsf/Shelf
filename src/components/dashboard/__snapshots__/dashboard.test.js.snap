// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    item={
      Array [
        Object {
          "id": 141,
          "img": "http://tineye.com/images/widgets/mona.jpg",
          "name": "Monnna",
          "price": 9988,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "dashboard",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "dashboard",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
